### PR TITLE
VMQ-426 support for Erlang 20

### PR DESCRIPTION
**Pull request by codeadict**
This is still **WIP**

### CODE REVIEW

File: apps/vmq_commons/.travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 language: erlang
 otp_release:
-    - 17.5
-    - 18.3
+    - 10.3
```
Comment (gleber) [inline apps/vmq_commons/.travis.yml]: Should be 19.3

Comment (codeadict) [inline apps/vmq_commons/.travis.yml]: ahh thanks, typo :)

**Comment by larshesel** [comment]
I've been thinking a bit more this and I think the better way to go would be to not replace gen_fsm with gen_statem just yet. In the long run we'd need to move there, but the general policy is to keep deprecated functionality around for two major releases, so there's no rush to move to gen_statem. So I think the easy and low risk path to get onto OTP20  would be to add some `{nowarn_deprecated_function, MFAs}` to the compile attribute in the relevant modules.

**Comment by codeadict** [comment]
Ahh yeah that sounds way easier, will do that for now, same for Plumtree and we will be ready to roll

**Comment by dergraf** [comment]
closing in favor of #450 . Thanks a ton for your efforts!

**Pull request closed without merging**